kodi_addon_checker/schema_validation.py

```diff
@@ -0,0 +1,47 @@
+import os
+import xmlschema
+import logging
+
+from .report import Report
+from .record import PROBLEM, Record, INFORMATION
+
+xml_schema = "./xml_schema/"
+LOGGER = logging.getLogger(__name__)
+
+
+def schemas(report: Report, parsed_xml):
+    metadatacount = 0
+    validcount = 1
+    for extension in parsed_xml.findall("extension"):
+        extension_point = (extension.get("point")).split(".")[-1]
```

Comment:
this fails at least for context menu extension points. Example:

´´´xml
  <extension point="kodi.context.item">
    <menu id="kodi.core.manage">
      <item library="addon.py">
        <label>32000</label>
        <visible>String.IsEqual(ListItem.dbtype,movie) | String.IsEqual(ListItem.dbtype,tvshow)</visible>
      </item>
    </menu>
  </extension>
```

Reply:
is it failing only for contextitem only ? If yes then I can add another if statement.
Something like
```python
if extension_point == "item":
   extension_point = "contextitem"
```

Reply:
There are also other problems you should take into account, this is not as easy as a new if statement. You need to create some sort of mapping between extension points and schemas. From kodi source tree:
```
  <extension-point id="script" schema="script.xsd"/>
  <extension-point id="subtitles" schema="script.xsd"/>
  <extension-point id="lyrics" schema="script.xsd"/>
  <extension-point id="weather" schema="script.xsd"/>
  <extension-point id="library" schema="script.xsd"/>
  <extension-point id="screensaver" schema="script.xsd"/>
  <extension-point id="plugin" schema="pluginsource.xsd"/>
  <extension-point id="context.item" schema="contextitem.xsd"/>
```

Reply:
After looking at this I think having a dictionary lookup would be nice